pylintrc

```diff
@@ -0,0 +1,417 @@
+ignore-long-lines=^\s*(# )?<?https?://\S+>?$
+
+# Number of spaces of indent required inside a hanging or continued line.
+indent-after-paren=4
+
+# String used as indentation unit. This is usually "    " (4 spaces) or "\t" (1
+# tab).
+indent-string='    '
+
+# Maximum number of characters on a single line.
+max-line-length=100
+
+# Maximum number of lines in a module.
+max-module-lines=1000
+
+# Allow the body of a class to be on the same line as the declaration if body
+# contains single statement.
+single-line-class-stmt=no
+
+# Allow the body of an if to be on the same line as the test if there is no
+# else.
+single-line-if-stmt=no
+
+[IMPORTS]
+
+# Allow explicit reexports by alias from a package __init__.
+allow-reexport-from-package=no
+
+# Allow wildcard imports from modules that define __all__.
+allow-wildcard-with-all=no
+
+# Force import order to recognize a module as part of a third party library.
+known-third-party=enchant
+
+[LOGGING]
+
+# The type of string formatting that logging methods do. `old` means using %
+# formatting, `new` is for `{}` formatting.
+logging-format-style=old
+
+# Logging modules to check that the string format arguments are in logging
+# function parameter format.
+logging-modules=logging
+
+
+[MESSAGES CONTROL]
+
+# Only show warnings with the listed confidence levels. Leave empty to show
+# all. Valid levels: HIGH, CONTROL_FLOW, INFERENCE, INFERENCE_FAILURE,
+# UNDEFINED.
+confidence=HIGH,
+           CONTROL_FLOW,
+           INFERENCE,
+           INFERENCE_FAILURE,
+           UNDEFINED
+
+# Disable the message, report, category or checker with the given id(s). You
+# can either give multiple identifiers separated by comma (,) or put this
+# option multiple times (only on the command line, not in the configuration
+# file where it should appear only once). You can also use "--disable=all" to
+# disable everything first and then re-enable specific checks. For example, if
+# you want to run only the similarities checker, you can use "--disable=all
+# --enable=similarities". If you want to run only the classes checker, but have
+# no Warning level messages displayed, use "--disable=all --enable=classes
+# --disable=W".
+disable=raw-checker-failed,
+        bad-inline-option,
+        locally-disabled,
+        file-ignored,
+        suppressed-message,
+        useless-suppression,
+        deprecated-pragma,
+        use-symbolic-message-instead,
+        use-implicit-booleaness-not-comparison-to-string,
+        use-implicit-booleaness-not-comparison-to-zero,
+        line-too-long,
+        invalid-name,
+        missing-module-docstring,
+        missing-function-docstring,
+        too-many-arguments,
+        too-many-locals,
+        unspecified-encoding,
+        too-many-return-statements,
+        too-many-branches,
+        too-many-nested-blocks,
+        too-many-instance-attributes,
+        too-many-statements,
+        too-few-public-methods,
+        fixme,
+        too-many-positional,
```

Comment:
Getting this error here:

```
pylintrc:1:0: W0012: Unknown option value for '--disable', expected a valid pylint message and got 'too-many-positional' (unknown-option-value)
```

Reply:
what pylint version you running? Looks like nixpkgs is on 3.1.1 (fairly far behind) which has this option, but 4.0.0 has `too-many-positional-arguments` [arg](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/too-many-positional-arguments.html). 4.0.0 would be the version we should follow in this case.

Reply:
nixpkgs `24.11` release is around the corner (currently in beta) and has 3.3.1. I'll test it.

Reply:
seems like I tested this outside of the nix shell and there I have pylint 3.3.1, so yeah, using that should fix it.

Reply:
updated and tested locally with 3.3.1 :+1: